server/command.go

```
@@ -0,0 +1,539 @@
+			}
+			newMessages = append(newMessages, message)
+		}
+		queue.Messages = newMessages
+		nErr := p.SaveQueues()
+		if nErr != nil {
+			p.API.LogError(nErr.Error())
+		}
+		_ = p.API.SendEphemeralPost(args.UserId, &model.Post{
+			ChannelId: args.ChannelId,
+			Message:   "Message inserted in the queue",
+		})
+		return &model.CommandResponse{}, nil
+	}
+
+	if split[1] == "list-messages" {
+		if len(split) < 3 {
+			_ = p.API.SendEphemeralPost(args.UserId, &model.Post{
+				ChannelId: args.ChannelId,
+				Message:   "Not enough arguments to list messages",
+			})
+			return &model.CommandResponse{}, nil
+		}
+		queue, ok := p.Queues[split[2]]
+		if !ok {
+			_ = p.API.SendEphemeralPost(args.UserId, &model.Post{
+				ChannelId: args.ChannelId,
+				Message:   fmt.Sprintf("Unknown queue %s.", split[2]),
+			})
+			return &model.CommandResponse{}, nil
+		}
+
+		listOfMessages := []string{fmt.Sprintf("#### List of messages for the queue %s:", queue.Name)}
+		for id, message := range queue.Messages {
+			listOfMessages = append(listOfMessages, fmt.Sprintf(" * **%d**: %s", id, message))
+		}
+		_ = p.API.SendEphemeralPost(args.UserId, &model.Post{
+			ChannelId: args.ChannelId,
+			Message:   strings.Join(listOfMessages, "\n"),
+		})
+		return &model.CommandResponse{}, nil
+	}
+	return &model.CommandResponse{}, nil
+}
+
+func (p *Plugin) executeDeferCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	split := strings.Fields(args.Command)
+	timeSpec := ""
+	if len(split) < 3 {
+		if len(split) == 2 && split[1] == "help" {
+			return p.executeDeferHelpCommand(c, args)
+		}
+
+		return &model.CommandResponse{
+				ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+				ChannelId:    args.ChannelId,
+				Text:         "Not enough parameters",
+			}, &model.AppError{
+				Message:       "Not enough parameters",
+				DetailedError: "Not enough parameters",
+			}
```

Do we really need to return an error here? I think that may get logged right? Either way, the user message seems like enough to me

---

Yes, you are right, I think it is enough.